load statements aren't handled

I recently tried to inline `tools.deps.alpha` to get hotload-dependencies working again for cljr, but mranderson failed to consider these [load statements](https://github.com/clojure/tools.deps.alpha/blob/99e95b7c709608c662092e3cb6fd3c8145c3afbd/src/main/clojure/clojure/tools/deps/alpha.clj#L816-L820) in the middle of the file.

Got an error:

```
unzipping [ plexus-component-annotations  [ v2v1v0 ]]
  munge source files of plexus-component-annotations artifact on branch [] exposed false.
  munge source files of commons-compress artifact on branch [] exposed false.
  munge source files of commons-lang artifact on branch [] exposed false.
  munge source files of xz artifact on branch [] exposed false.
  munge source files of compliment artifact on branch [] exposed false.
Error encountered performing task 'inline-deps' with profile(s): 'base,system,provided,dev,1.10'
java.util.concurrent.ExecutionException: clojure.lang.ExceptionInfo: Invalid symbol: Error:. {:type :reader-exception, :ex-kind :reader-error}
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at clojure.core$deref_future.invokeStatic(core.clj:2300)
	at clojure.core$future_call$reify__8439.deref(core.clj:6974)
	at clojure.core$deref.invokeStatic(core.clj:2320)
	at clojure.core$pmap$step__8452$fn__8456.invoke(core.clj:7025)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:709)
	at clojure.core$next__5371.invokeStatic(core.clj:64)
	at clojure.core$dorun.invokeStatic(core.clj:3142)
	at clojure.core$doall.invokeStatic(core.clj:3148)
	at clojure.core$doall.invoke(core.clj:3148)
	at mranderson.move$replace_ns_symbol_in_source_files.invokeStatic(move.clj:303)
	at mranderson.move$replace_ns_symbol_in_source_files.invoke(move.clj:294)
	at mranderson.move$move_ns.invokeStatic(move.clj:318)
	at mranderson.move$move_ns.invoke(move.clj:305)
	at mranderson.core$update_artifact_BANG_.invokeStatic(core.clj:227)
	at mranderson.core$update_artifact_BANG_.invoke(core.clj:196)
	at mranderson.dependency.tree$walk_ordered_deps$fn__2194$fn__2199.invoke(tree.clj:83)
	at clojure.core$map$fn__5851.invoke(core.clj:2755)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:709)
	at clojure.core$next__5371.invokeStatic(core.clj:64)
	at clojure.core$dorun.invokeStatic(core.clj:3142)
	at clojure.core$dorun.invoke(core.clj:3133)
	at mranderson.dependency.tree$walk_ordered_deps$fn__2194.invoke(tree.clj:83)
	at mranderson.dependency.tree$walk_ordered_deps.invokeStatic(tree.clj:82)
	at mranderson.dependency.tree$walk_ordered_deps.invoke(tree.clj:73)
	at mranderson.core$mranderson_resolved_deps_BANG_.invokeStatic(core.clj:326)
	at mranderson.core$mranderson_resolved_deps_BANG_.invoke(core.clj:307)
	at mranderson.core$mranderson.invokeStatic(core.clj:357)
	at mranderson.core$mranderson.invoke(core.clj:333)
	at leiningen.inline_deps$inline_deps.invokeStatic(inline_deps.clj:65)
	at leiningen.inline_deps$inline_deps.doInvoke(inline_deps.clj:52)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at leiningen.core.main$partial_task$fn__6592.doInvoke(main.clj:284)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at leiningen.core.main$apply_task.invokeStatic(main.clj:334)
	at leiningen.core.main$apply_task.invoke(main.clj:320)
	at leiningen.with_profile$with_profiles_STAR_.invokeStatic(with_profile.clj:14)
	at leiningen.with_profile$with_profiles_STAR_.invoke(with_profile.clj:8)
	at leiningen.with_profile$apply_task_with_profiles.invokeStatic(with_profile.clj:53)
	at leiningen.with_profile$apply_task_with_profiles.invoke(with_profile.clj:45)
	at leiningen.with_profile$with_profile$fn__10482.invoke(with_profile.clj:85)
	at clojure.core$mapv$fn__8430.invoke(core.clj:6912)
	at clojure.core.protocols$fn__8144.invokeStatic(protocols.clj:168)
	at clojure.core.protocols$fn__8144.invoke(protocols.clj:124)
	at clojure.core.protocols$fn__8099$G__8094__8108.invoke(protocols.clj:19)
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
	at clojure.core.protocols$fn__8131.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8131.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8073$G__8068__8086.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6828)
	at clojure.core$mapv.invokeStatic(core.clj:6903)
	at clojure.core$mapv.invoke(core.clj:6903)
	at leiningen.with_profile$with_profile.invokeStatic(with_profile.clj:85)
	at leiningen.with_profile$with_profile.doInvoke(with_profile.clj:63)
	at clojure.lang.RestFn.invoke(RestFn.java:445)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at leiningen.core.main$partial_task$fn__6592.doInvoke(main.clj:284)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at leiningen.core.main$apply_task.invokeStatic(main.clj:334)
	at leiningen.core.main$apply_task.invoke(main.clj:320)
	at leiningen.core.main$resolve_and_apply.invokeStatic(main.clj:343)
	at leiningen.core.main$resolve_and_apply.invoke(main.clj:336)
	at leiningen.core.main$_main$fn__6681.invoke(main.clj:452)
	at leiningen.core.main$_main.invokeStatic(main.clj:442)
	at leiningen.core.main$_main.doInvoke(main.clj:439)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.main$main_opt.invokeStatic(main.clj:491)
	at clojure.main$main_opt.invoke(main.clj:487)
	at clojure.main$main.invokeStatic(main.clj:598)
	at clojure.main$main.doInvoke(main.clj:561)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.main.main(main.java:37)
Caused by: clojure.lang.ExceptionInfo: Invalid symbol: Error:. {:type :reader-exception, :ex-kind :reader-error}
	at clojure.tools.reader.impl.errors$throw_ex.invokeStatic(errors.clj:34)
	at clojure.tools.reader.impl.errors$throw_ex.doInvoke(errors.clj:24)
	at clojure.lang.RestFn.invoke(RestFn.java:442)
	at clojure.tools.reader.impl.errors$reader_error.invokeStatic(errors.clj:40)
	at clojure.tools.reader.impl.errors$reader_error.doInvoke(errors.clj:36)
	at clojure.lang.RestFn.invoke(RestFn.java:516)
	at clojure.tools.reader.impl.errors$throw_invalid.invokeStatic(errors.clj:97)
	at clojure.tools.reader.impl.errors$throw_invalid.invoke(errors.clj:96)
	at clojure.tools.reader.edn$read_symbol.invokeStatic(edn.clj:262)
	at clojure.tools.reader.edn$read_symbol.invoke(edn.clj:249)
	at clojure.tools.reader.edn$read.invokeStatic(edn.clj:407)
	at clojure.tools.reader.edn$read.invoke(edn.clj:371)
	at clojure.tools.reader.edn$read.invokeStatic(edn.clj:392)
	at clojure.tools.reader.edn$read.invoke(edn.clj:371)
	at clojure.tools.reader.edn$read_string.invokeStatic(edn.clj:440)
	at clojure.tools.reader.edn$read_string.invoke(edn.clj:429)
	at clojure.tools.reader.edn$read_string.invokeStatic(edn.clj:437)
	at clojure.tools.reader.edn$read_string.invoke(edn.clj:429)
	at rewrite_clj.reader$string__GT_edn.invokeStatic(reader.cljc:109)
	at rewrite_clj.reader$string__GT_edn.invoke(reader.cljc:106)
	at rewrite_clj.parser.token$symbol_node.invokeStatic(token.cljc:34)
	at rewrite_clj.parser.token$symbol_node.invoke(token.cljc:23)
	at rewrite_clj.parser.token$parse_token.invokeStatic(token.cljc:47)
	at rewrite_clj.parser.token$parse_token.invoke(token.cljc:37)
	at rewrite_clj.parser.core$eval4078$fn__4079.invoke(core.cljc:74)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at rewrite_clj.reader$read_with_meta.invokeStatic(reader.cljc:147)
	at rewrite_clj.reader$read_with_meta.invoke(reader.cljc:143)
	at rewrite_clj.parser.core$parse_next.invokeStatic(core.cljc:46)
	at rewrite_clj.parser.core$parse_next.invoke(core.cljc:44)
	at rewrite_clj.parser.core$parse_delim$fn__4070.invoke(core.cljc:54)
	at rewrite_clj.reader$read_repeatedly$fn__3108.invoke(reader.cljc:156)
	at clojure.core$repeatedly$fn__6448.invoke(core.clj:5149)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:531)
	at clojure.core$seq__5387.invokeStatic(core.clj:137)
	at clojure.core$take_while$fn__5902.invoke(core.clj:2904)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:709)
	at clojure.core$next__5371.invokeStatic(core.clj:64)
	at clojure.core$dorun.invokeStatic(core.clj:3142)
	at clojure.core$doall.invokeStatic(core.clj:3148)
	at clojure.core$doall.invoke(core.clj:3148)
	at rewrite_clj.reader$read_repeatedly.invokeStatic(reader.cljc:158)
	at rewrite_clj.reader$read_repeatedly.invoke(reader.cljc:152)
	at rewrite_clj.parser.core$parse_delim.invokeStatic(core.cljc:55)
	at rewrite_clj.parser.core$parse_delim.invoke(core.cljc:50)
	at rewrite_clj.parser.core$eval4148$fn__4149.invoke(core.cljc:201)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at rewrite_clj.reader$read_with_meta.invokeStatic(reader.cljc:147)
	at rewrite_clj.reader$read_with_meta.invoke(reader.cljc:143)
	at rewrite_clj.parser.core$parse_next.invokeStatic(core.cljc:46)
	at rewrite_clj.parser.core$parse_next.invoke(core.cljc:44)
	at rewrite_clj.reader$read_n.invokeStatic(reader.cljc:168)
	at rewrite_clj.reader$read_n.invoke(reader.cljc:160)
	at rewrite_clj.parser.core$parse_printables.invokeStatic(core.cljc:61)
	at rewrite_clj.parser.core$parse_printables.doInvoke(core.cljc:57)
	at clojure.lang.RestFn.invoke(RestFn.java:445)
	at rewrite_clj.parser.core$eval4110$fn__4111.invoke(core.cljc:118)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at rewrite_clj.reader$read_with_meta.invokeStatic(reader.cljc:147)
	at rewrite_clj.reader$read_with_meta.invoke(reader.cljc:143)
	at rewrite_clj.parser.core$parse_next.invokeStatic(core.cljc:46)
	at rewrite_clj.parser.core$parse_next.invoke(core.cljc:44)
	at rewrite_clj.parser.core$parse_delim$fn__4070.invoke(core.cljc:54)
	at rewrite_clj.reader$read_repeatedly$fn__3108.invoke(reader.cljc:156)
	at clojure.core$repeatedly$fn__6448.invoke(core.clj:5149)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.RT.seq(RT.java:531)
	at clojure.core$seq__5387.invokeStatic(core.clj:137)
	at clojure.core$take_while$fn__5902.invoke(core.clj:2904)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:709)
	at clojure.core$next__5371.invokeStatic(core.clj:64)
	at clojure.core$dorun.invokeStatic(core.clj:3142)
	at clojure.core$doall.invokeStatic(core.clj:3148)
	at clojure.core$doall.invoke(core.clj:3148)
	at rewrite_clj.reader$read_repeatedly.invokeStatic(reader.cljc:158)
	at rewrite_clj.reader$read_repeatedly.invoke(reader.cljc:152)
	at rewrite_clj.parser.core$parse_delim.invokeStatic(core.cljc:55)
	at rewrite_clj.parser.core$parse_delim.invoke(core.cljc:50)
	at rewrite_clj.parser.core$eval4140$fn__4141.invoke(core.cljc:193)
	at clojure.lang.MultiFn.invoke(MultiFn.java:229)
	at rewrite_clj.reader$read_with_meta.invokeStatic(reader.cljc:147)
	at rewrite_clj.reader$read_with_meta.invoke(reader.cljc:143)
	at rewrite_clj.parser.core$parse_next.invokeStatic(core.cljc:46)
	at rewrite_clj.parser.core$parse_next.invoke(core.cljc:44)
	at rewrite_clj.parser$parse.invokeStatic(parser.cljc:20)
	at rewrite_clj.parser$parse.invoke(parser.cljc:17)
	at mranderson.move$split_ns_form_ns_body.invokeStatic(move.clj:139)
	at mranderson.move$split_ns_form_ns_body.invoke(move.clj:130)
	at mranderson.move$replace_ns_symbol.invokeStatic(move.clj:256)
	at mranderson.move$replace_ns_symbol.invoke(move.clj:247)
	at clojure.lang.AFn.applyToHelper(AFn.java:178)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at mranderson.move$update_file.invokeStatic(move.clj:38)
	at mranderson.move$update_file.doInvoke(move.clj:32)
	at clojure.lang.RestFn.invoke(RestFn.java:566)
	at mranderson.move$replace_ns_symbol_in_source_files$fn__4897.invoke(move.clj:302)
	at clojure.core$pmap$fn__8447$fn__8448.invoke(core.clj:7022)
	at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2030)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
make: *** [Makefile:10: .inline-deps] Error 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
 
 ## Bug fixes
 
+- [fix [#61](https://github.com/benedekfazekas/mranderson/issues/61)] Repoint `(load "…")` resource paths when a namespace is relocated, so libraries that `load` companion files (e.g. `tools.deps.alpha`) inline correctly
 - [fix [#73](https://github.com/benedekfazekas/mranderson/issues/73)] Munge the package of fully-qualified class/record references to Java style so a dash in the prefix doesn't produce broken references
 - [fix [#64](https://github.com/benedekfazekas/mranderson/issues/64)] Strip non-alphanumeric characters (e.g. a `/` in a `n/a` version) from the generated prefix so they don't break imports
 - [fix [#54](https://github.com/benedekfazekas/mranderson/issues/54)] [fix [#92](https://github.com/benedekfazekas/mranderson/issues/92)] Rewrite Java `:import`s of inlined classes in top-level dependency namespaces and in the consuming project's own sources, which the per-dependency pass missed

--- a/src/mranderson/move.clj
+++ b/src/mranderson/move.clj
@@ -70,6 +70,15 @@
 (defn- java-package [sym]
   (str/replace (name sym) "-" "_"))
 
+(defn- load-param
+  "Renders `sym` as the resource path used by `(load \"…\")`: dots become
+  slashes and dashes underscores, e.g. `clojure.tools.deps.alpha` ->
+  `clojure/tools/deps/alpha`."
+  [sym]
+  (-> (name sym)
+      (str/replace "." "/")
+      (str/replace "-" "_")))
+
 (defn- java-style-prefix?
   [old-sym node]
   (when-not (#{:uneval} (z/tag node))
@@ -259,7 +268,9 @@
         old-pkg-prefix  (java-package old-sym)
         new-pkg-prefix  (java-package new-sym)
         old-type-prefix (str "^" (java-package old-sym))
-        new-type-prefix (str "^" (java-package new-sym))]
+        new-type-prefix (str "^" (java-package new-sym))
+        old-load-param  (load-param old-sym)
+        new-load-param  (load-param new-sym)]
     (cond
 
       (= match old-ns-ref)
@@ -281,6 +292,10 @@
           (java-package replaced)
           replaced))
 
+      ;; a `(load "…")` resource path, in slash/underscore form. See #61.
+      (str/starts-with? match old-load-param)
+      (str/replace match old-load-param new-load-param)
+
       :default
       match)))
 
@@ -291,7 +306,12 @@
   #"\"?[a-zA-Z0-9$%*+=?!<>^_-]['.a-zA-Z0-9$%*+=?!<>_-]*")
 
 (defn- replace-in-source [source-sans-ns old-sym new-sym]
-  (str/replace source-sans-ns symbol-regex (partial source-replacement old-sym new-sym)))
+  ;; `symbol-regex` never matches a `(load "…")` resource path (it stops at `/`),
+  ;; so try the slash/underscore form of the namespace first. See #61.
+  (let [regex (re-pattern (str (java.util.regex.Pattern/quote (load-param old-sym))
+                               "|"
+                               (.pattern ^java.util.regex.Pattern symbol-regex)))]
+    (str/replace source-sans-ns regex (partial source-replacement old-sym new-sym))))
 
 (defn- after-platform-marker? [platform node]
   (when-not (#{:uneval} (z/tag node))

--- a/test-resources/load-statement-example.txt
+++ b/test-resources/load-statement-example.txt
@@ -1,0 +1,16 @@
+(ns
+  ^{:author "Alex Taggart"
+    :doc
+    "Support for testing whether logging calls are made.
+
+  Usage example:
+    (require '[clojure.tools.logging :as log]
+             '[clojure.tools.logging.test :refer [logged? with-log])
+
+    (with-log
+      (log/info \"Hello World!\")
+      (log/error (Exception. \"Did a thing\") \"Error: oops\")
+      (logged? 'user :info #\"Hello\"))"}
+  foo.bar.alpha)
+
+(load "/foo/bar/alpha/extensions/maven")

--- a/test/mranderson/move_test.clj
+++ b/test/mranderson/move_test.clj
@@ -304,6 +304,29 @@
       (t/is (= (slurp file-moved-metamap2) expected-moved-metamap2-watermark)))))
 
 
+(t/deftest move-ns-load-statement-test
+  ;; `(load "…")` embeds a resource PATH (slashes), which the symbol-based body
+  ;; rewriting never sees; it must be repointed when the namespace moves (#61).
+  ;; The namespace also carries a docstring full of example Clojure code and
+  ;; escaped quotes, to confirm that doesn't trip up ns-form parsing.
+  (let [temp-dir (create-temp-dir! "mranderson-load")
+        src-dir  (io/file temp-dir "src")
+        _        (create-source-file! (io/file src-dir "foo" "bar" "alpha.clj")
+                                      (slurp "test-resources/load-statement-example.txt"))
+        moved    (io/file src-dir "moved" "foo" "bar" "alpha.clj")]
+    (sut/move-ns 'foo.bar.alpha 'moved.foo.bar.alpha src-dir ".clj" [src-dir] nil)
+    (t/is (.exists moved)
+          "the file is moved to the new namespace location")
+    (let [out (slurp moved)]
+      (t/is (.contains out "moved.foo.bar.alpha")
+            "the namespace is renamed")
+      (t/is (.contains out "(load \"/moved/foo/bar/alpha/extensions/maven\")")
+            "the load path is repointed to the new location")
+      (t/is (not (.contains out "\"/foo/bar/alpha/extensions"))
+            "the original load path is gone")
+      (t/is (.contains out "Alex Taggart")
+            "the docstring metadata is preserved through the move"))))
+
 (defn- s-rename-ns
   "A litle helper for rename-ns-test"
   [s old-ns-name new-ns-name add-meta-kw]
@@ -373,4 +396,9 @@
 
       ;; the bare namespace itself keeps the dashes
       "cider.nrepl.inlined-deps.instaparse.gll"
-      "instaparse.gll")))
+      "instaparse.gll"
+
+      ;; a (load "…") resource path is repointed in slash/underscore form (#61);
+      ;; the dash in the prefix becomes an underscore, like a package path
+      "(load \"/cider/nrepl/inlined_deps/instaparse/gll/extra\")"
+      "(load \"/instaparse/gll/extra\")")))

